DatadogSessionReplay/Sources/DatadogSessionReplay/Writer/Models/EnrichedRecord.swift

```diff
@@ -58,8 +58,8 @@ internal struct EnrichedRecord: Encodable {
         }
 
         self.hasFullSnapshot = hasFullSnapshot
-        self.earliestTimestamp = earliestTimestamp
-        self.latestTimestamp = latestTimestamp
+        self.earliestTimestamp = earliestTimestamp &- rumContext.serverTimeOffset
+        self.latestTimestamp = latestTimestamp &- rumContext.serverTimeOffset
```

Comment:
Applying the offset in this place doesn't make much sense - we only alter the `EnrichedRecord` which is a utility type used later to build `Segment`. In result, the segment will have correct "start" and "end" time, but all records in it will be still skewed (notice that this change ☝️ doesn't alter `self.records`).

If we apply the offset early in the `Recorder` layer - it will be automatically leveraged everywhere else (in `Processor`, `Writer` and finally `RequestBuilder`). There are two TODOs that indicate good strategic right places for date correction:

https://github.com/DataDog/dd-sdk-ios/blob/8dbd3472e52ac7156c45987f5796e47b079762c8/DatadogSessionReplay/Sources/DatadogSessionReplay/Recorder/TouchSnapshotProducer/WindowTouchSnapshotProducer.swift#L53

https://github.com/DataDog/dd-sdk-ios/blob/8dbd3472e52ac7156c45987f5796e47b079762c8/DatadogSessionReplay/Sources/DatadogSessionReplay/Recorder/Recorder.swift#L136